TreeDB/docs/evidence/vector-partition-locality-matrix-4142/tools/test_reduce_matrix.py

```
@@ -0,0 +1,139 @@
+            def write(name: str, value: dict) -> Path:
+                path = root / name
+                path.write_text(json.dumps(value), encoding="utf-8")
+                return path
+            preflight_path = write("preflight.json", preflight())
+            paths = [write(f"{value['row_id']}.json", value) for value in complete_rows()]
+            self.assertEqual(reducer.reduce_rows(paths, preflight_path)["rows"], len(reducer.AUTHORIZED_COORDINATES))
+            summary_path = root / "summary.json"
+            with mock.patch.object(sys, "argv", ["reduce_matrix.py", "--preflight", str(preflight_path), "--out", str(summary_path), *map(str, paths)]):
+                reducer.main()
+            self.assertEqual(json.loads(summary_path.read_text(encoding="utf-8"))["rows"], len(reducer.AUTHORIZED_COORDINATES))
+            with self.assertRaisesRegex(reducer.ContractError, "incomplete"):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)